share/picongpu/tests/Synchrotron_plugin/lib/python/test/SynchrotronRadiation/validate.py

```diff
@@ -0,0 +1,174 @@
+import sys
+import openpmd_api as io
+import numpy as np
+import scipy.constants as const
+from scipy.integrate import quad
+from scipy.special import kv
+
+e = const.elementary_charge  # Elementary charge in C
+m_e = const.m_e  # Electron mass in kg
+c = const.c  # Speed of light in m/s
+hbar = const.hbar  # Reduced Planck constant in J/s
+
+miu0 = const.mu_0  # Vacuum permeability in H/m
+eps0 = const.epsilon_0  # Vacuum permittivity in F/m
+Es = m_e**2 * c**3 / e / hbar  # Schwinger limit in V/m
+
+
+def F1(z_q):
+    if z_q > 2.9e-6:
+        integral = quad(lambda x: kv(5 / 3, x), z_q, np.inf)[0]
+        return z_q * integral
+    else:
+        return 2.15 * z_q ** (1 / 3)
+
+
+def F2(z_q):
+    return z_q * kv(2 / 3, z_q)
+
+
+def analytical_Propability(delta, gamma, Heff, dt):
+    chi = gamma * Heff / Es
+    zq = 2 / (3 * chi) * delta / (1 - delta)
+    F1_result = F1(zq)
+    F2_result = F2(zq)
+    numerator = (
+        dt * e**2 * m_e * c * np.sqrt(3) * chi * (1 - delta) * (F1_result + 3 * delta * zq * chi / 2 * F2_result)
+    )
+    denominator = 2 * np.pi * gamma * delta * hbar**2 * eps0 * 4 * np.pi
+    return numerator / denominator
+
+
+def electron_energy(gamma):
+    return gamma * (m_e * c**2 * 6.2415e18)  # Energy of the electrons in eV
+
+
+# interpolate y1 to x0 and subtract y0 from y1
+def subtract_functions(x0, y0, x1, y1):
```

Comment:
please add a docstring that explains what the function does and what arguments it expects

Reply:
done
``` 
# Used when comparing two sets of histogram data with different binning
# given two set of vectors x0, y0 and x1, y1 we interpolate second set to the first set and subtract them
# interpolate y1 to x0 and subtract y0 from y1 
```